src/misc/msccl/msccl_lifecycle.cc

```diff
@@ -592,6 +592,7 @@ ncclResult_t mscclEnqueueCheck(
         }
       threadLocalStatus.groupStatus = mscclGroupUnsupportedOp;
       NCCLCHECK(mscclFallBackSavedParams());
+      break;
     case mscclGroupUnsupportedOp:
       NCCLCHECK(mscclFallBackSavedParams());
```

Comment:
Alternatively, is this fall-through intentional? If so, it double calls mscclFallBackSavedParams. Is that also intentional?

Reply:
The call to `threadLocalStatus.savedSchedulerParams.clear();` at the end of the fallback function tells me it is unintentional. No one tries to refill the saveSchedulerParams in between as it is thread_local.